biorxiv/publication_delay_experiment/03_publication_delay_experiment_reviewer_request.py

```diff
@@ -0,0 +1,122 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.9.1+dev
+#   kernelspec:
+#     display_name: Python [conda env:annorxiver]
+#     language: python
+#     name: conda-env-annorxiver-py
+# ---
+
+# # Measure the Difference between Preprint-Published similarity and Published Articles
+
+# This notebook is designed to output an excel file that contains all preprint published pairs and their respective publication information.
+
+from datetime import timedelta
+import numpy as np
+import pandas as pd
+import plotnine as p9
+from scipy.stats import linregress
+
+# # Load the Document Distances
+
+published_date_distances = pd.read_csv(
+    "output/preprint_published_distances.tsv", sep="\t"
+)
+for col in ["preprint_date", "published_date"]:
+    published_date_distances[col] = pd.to_datetime(published_date_distances[col])
+published_date_distances["time_to_published"] = pd.to_timedelta(
+    published_date_distances["time_to_published"]
+)
+print(published_date_distances.shape)
+published_date_distances.head()
+
+# # Plot Version count against Doc Distances
+
+# Reviewer wanted to see if there is an association between version count and document distances. (i.e. if preprints with more versions have more text changes).
+
+# +
+x = (published_date_distances["version_count"].values.tolist(),)
+y = published_date_distances["doc_distances"].values.tolist()
+
+results = linregress(x, y)
+print(results)
+# -
+
+published_date_distances["version_count"] = pd.Categorical(
+    published_date_distances["version_count"].tolist()
+)
+g = (
+    p9.ggplot(published_date_distances, p9.aes(x="version_count", y="doc_distances"))
+    + p9.geom_boxplot(fill="#b2df8a")
+    + p9.geom_line(
+        data=pd.DataFrame(
+            dict(
+                version_count=np.arange(1, 13),
+                doc_distances=np.arange(1, 13) * 0.02669 + 0.8697,
+            )
+        ),
+        linetype="dashed",
+        color="#1f78b4",
+        size=1,
+    )
+    + p9.annotate(
+        "text",
+        label=f"y={results.slope:0.4f}*X + {results.intercept:0.4f}",
+        x=9,
+        y=7.5,
+        size=13,
+        color="#1f78b4",
+    )
+    + p9.labs(
+        x="# of Preprint Versions",
+        y="Euclidean Distance of Preprint-Published Versions",
+    )
+    + p9.theme_seaborn(style="white", context="notebook")
```

Comment:
What is the takeaway for this plot? It looks like if you remove the "outliers" that you might have a nice positive correlation between the distance and the number of versions, which I would intuitively expect more text changes with more version and thus a large difference between the final and original version. However, I'm not sure what these "outlier"-looking documents are so I wouldn't want to suggest removing them. I'm more curious what your thoughts are here.

Reply:
>What is the takeaway for this plot? It looks like if you remove the "outliers" that you might have a nice positive correlation between the distance and the number of versions, which I would intuitively expect more text changes with more version and thus a large difference between the final and original version. However, I'm not sure what these "outlier"-looking documents are so I wouldn't want to suggest removing them. I'm more curious what your thoughts are here.

So basically this plot shows a weak but positive relationship between number of preprint versions and document embedding distances. Based on the slope it says that for every preprint version the document distance changes by 0.028. So preprints are being changed with each new version. If you compared that slope to the distance between a random preprint and a published article (~ 6 distance units), then you can say that these "changes" are minor relative to a full article change. This goes against a previous assumption that each new preprint version should bring substantial changes, which is interesting. Overall, preprints do change with each new version. Just not to the extent as one, without prior knowledge about publishing and preprints, would expect.